analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -272,15 +272,27 @@ public void Execute_ClearsCapturesAfterBranching()
 
     [TestMethod]
     public void Execute_LocalScopeRegion_Boolean_AssignDefaultBoolConstraint() =>
-        SETestContext.CreateVB(@"Dim Value As Boolean : Tag(""Value"", Value)").Validator.ValidateTag("Value", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
+        SETestContext.CreateVB(@"Dim Value As Boolean : Tag(""Value"", Value)").Validator.TagValue("Value").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.False);
 
     [TestMethod]
     public void Execute_LocalScopeRegion_ReferenceType_AssignDefaultNullConstraint() =>
-        SETestContext.CreateVB(@"Dim Value As Exception : Tag(""Value"", Value)").Validator.ValidateTag("Value", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        SETestContext.CreateVB(@"Dim Value As Exception : Tag(""Value"", Value)").Validator.TagValue("Value").Should().HaveOnlyConstraint(ObjectConstraint.Null);
+
+    [DataTestMethod]
+    [DataRow("SByte")]
+    [DataRow("Byte")]
+    [DataRow("Short")]
+    [DataRow("UShort")]
+    [DataRow("Integer")]
+    [DataRow("UInteger")]
+    [DataRow("Long")]
+    [DataRow("ULong")]
+    public void Execute_LocalScopeRegion_Integral_NoAction(string type) =>
```

Comment:
This fails with `System.IntPtr` and `System.UIntPtr`, which are considered integral types by the [language reference](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types#characteristics-of-the-integral-types), but not by `KnownType.IntegralNumbers`.
```cs
        public static readonly ImmutableArray<KnownType> IntegralNumbers =
            ImmutableArray.Create(
                System_Int16,
                System_Int32,
                System_Int64,
                System_UInt16,
                System_UInt32,
                System_UInt64,
                System_Char,
                System_Byte,
                System_SByte);
```

Since the 0 pointer makes sense and checking for a pointer to be assigned before using it is quite common, an unassigned `System.IntPtr` should have the `NumberConstraint.From(0)`.

Otherwise, code like the following would make wrong conclusions about the flow:
```vbnet
Dim ptr As System.IntPtr ' Initialized to 0
If SomethingAlwaysNegative > 0 Then
    ptr = SomethingValid
End If
If ptr = 0 Then
    Throw ...
End If
```

I think it's enough to change `ConstraintFromType` into the following:
```cs
        public static SymbolicConstraint ConstraintFromType(ITypeSymbol type)
        {
            if (type.Is(KnownType.System_Boolean))
            {
                return BoolConstraint.False;
            }
            else if (type.IsReferenceType)
            {
                return ObjectConstraint.Null;
            }
            else if (type.IsAny(KnownType.IntegralNumbersIncludingNative)) // Change is here
            {
                return NumberConstraint.From(0);
            }
            else
            {
                return null;
            }
        }
```

Reply:
The `_NoAction` name was actually wrong (copy/paste artifact) :D

Reply:
The `ptr = 0` required one more change, because it has custom conversion operator on the type